etc/scripts/docenizer.py

```diff
@@ -170,12 +171,12 @@ def parse_html(directory):
 """)
         for inst in instructions:
             for name in inst.names:
-                f.write('    case "{}":\n'.format(name))
-            f.write('        return {};\n\n'.format(json.dumps({
+                f.write('        case "{}":\n'.format(name))
+            f.write('            return {}'.format(json.dumps({
                 "tooltip": inst.tooltip,
                 "html": inst.body,
                 "url": "http://www.felixcloutier.com/x86/{}.html".format(inst.name)
```

Comment:
I just ran a quick check against every URL in the file, and I found 1 issue: Filenames are not url-encoded. 
An example: `http://www.felixcloutier.com/x86/FCOMI:FCOMIP: FUCOMI:FUCOMIP.html raised exception Bad Request` note how there's an space there.

Reply:
Oh, indeed. I'll come with a way of testing first and will use this as an example of failing test.

Reply:
I think the tests should be run only after generating the docs, not as part of the automated testing env that runs before every commit/build. This would make checking every url acceptable. In case you want to save some time, the awful script I used to check for every url can be found here https://gist.github.com/RabsRincon/4c8e41668161060b6a57b3e21ed41190

Reply:
Thanks for asking! At which frequency do you guys typically regenerate documentation? (I realize that you don't control your 400 forks, of course.)